Dip/DipTests/ComponentScopeTests.swift

```diff
@@ -90,14 +90,14 @@ class ComponentScopeTests: XCTestCase {
 
   func testThatItReusesInstanceInObjectGraphScopeDuringResolve() {
     //given
-    container.register(.ObjectGraph) { [unowned container] in Client(server: container.resolve()) as Client }
+    container.register(.ObjectGraph) { [unowned container] in Client(server: try! container.resolve()) as Client }
     
     container.register(.ObjectGraph) { Server() as Server }.resolveDependencies { container, server in
```

Comment:
> I don't think so, cause here we don't capture container from outer context, we pass it as an argument at runtime. So the reference should be released as soon as block returns.

Don't we? I mean, `container` captures `resolveDependenciesBlock` right? So unless that block don't use the `container` argument in the closure, this `container` is captured automatically by the closure and not "released as soon as the block returns" (as the `resolveDependenciesBlock` will still be captured and kept as a property in the `container`). So this captured `container` will only be released when the block itself is released (set back to `nil`).

(If the block were not stored as a strong property by the container but used right away, we would be able to mark it `@noescape`, which is not the case here)